assets/javascripts/kitten/components/text-copy/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { useRef } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import COLORS from '../../constants/colors-config'
+import { pxToRem } from '../../helpers/utils/typography'
+import { CopyIcon } from '../icons/copy-icon'
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  border: ${pxToRem(2)} solid ${COLORS.line1};
+  background-color: ${COLORS.background1};
+  cursor: pointer;
+`
+
+const StyledText = styled.div`
+  padding: ${pxToRem(10)} ${pxToRem(15)};
+`
+
+const IconWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  padding: ${pxToRem(10)};
+  border-left: ${pxToRem(2)} solid ${COLORS.line1};
+  align-self: stretch;
+  box-sizing: border-box;
+`
+
+export const TextCopy = ({ children, textToCopy }) => {
+  const textRef = useRef(null)
+  return (
+    <Wrapper
+      onClick={() => {
+        if (textToCopy) {
+          const el = document.createElement('textarea')
+          el.value = textToCopy
+          document.body.appendChild(el)
+          el.select()
+          document.execCommand('copy')
+          document.body.removeChild(el)
+          const range = document.createRange()
+          range.selectNode(textRef.current)
+          window.getSelection().addRange(range)
+        } else {
+          const range = document.createRange()
+          range.selectNode(textRef.current)
+          window.getSelection().addRange(range)
+          document.execCommand('copy')
+        }
+      }}
+    >
+      <StyledText ref={textRef}>{children}</StyledText>
```

Comment:
Si il n'y a rien indiqué autour du composant, juste avant le texte je rajouterais un `VisuallyHidden` avec un texte qui dit "Cliquez sur ce bouton pour copier le texte suivant : " pour les lecteurs d'écran.
```suggestion
      <VisuallyHidden>Cliquez sur ce bouton pour copier le texte suivant : </VisuallyHidden>
      <StyledText ref={textRef}>{children}</StyledText>
```

Reply:
Ok 👍 
Petite réflexion : Dommage que kitten ne fasse pas d'i18n en interne.